### PR TITLE
adds first draft of assembly actions

### PR DESCRIPTION
**Pull request by gregcaporaso**
There are still several to-do items in here, including addition of single-end read workflows and unit tests. Pushing this up now for possible merge so @nbokulich can work on a high-level `Pipeline` integrating this with the quality control workflow.

### CODE REVIEW

**Comment by thermokarst** [comment]
Side note - the github action appears to be reporting success, even though there are build/test failures. I am sorting out now.

**Comment by thermokarst** [comment]
Okay, I think I sorted out the erroneous passing build status - errors should signal a failing condition. Next up - force conda to resolve on the latest and greatest dev builds (right now it is occasionally snaggling an older build of q2-types 2020.5.0.dev0, rather than the _latest_ dev build).

**Comment by thermokarst** [comment]
Head's up @gregcaporaso - I updated this branch to reflect the latest changes in master (looks like you were a few commits behind when you cut this branch).

File: q2_phylogenomics/_assemble.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+from ._util import run_command
+from ._format import (
+        BAMFilesDirFmt, PileUpFilesDirFmt,
+        BAMFormat, PileUpTSVFormat)
+
+
+def map_paired_reads(demux: SingleLanePerSamplePairedEndFastqDirFmt,
+                     reference: Bowtie2IndexDirFmt,
+                     mismatches_per_seed: int = 1,
+                     ceil_func: str = '0,0.3',
```
Comment (nbokulich) [inline q2_phylogenomics/_assemble.py]: the bowtie2 --n-ceil option looks really flexible, more than we probably need here. Looks like you are already always using a linear function, so maybe also just use a linear intercept? In which case you can change this to a `float` and Q2 can handle the validation.

File: q2_phylogenomics/_assemble.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+from ._util import run_command
+from ._format import (
+        BAMFilesDirFmt, PileUpFilesDirFmt,
+        BAMFormat, PileUpTSVFormat)
+
+
+def map_paired_reads(demux: SingleLanePerSamplePairedEndFastqDirFmt,
+                     reference: Bowtie2IndexDirFmt,
+                     mismatches_per_seed: int = 1,
```
Comment (nbokulich) [inline q2_phylogenomics/_assemble.py:30]: want to make a dict of default params for each method? Then I can easily import that dict for setting defaults in the pipeline.

**Comment by thermokarst** [comment]
Hey @gregcaporaso - just a reminder to fetch changes from your remote branch - I have pushed a few tweaks up there to resolve the package building issues.

**Review by nbokulich — APPROVED** [review]
Thanks @gregcaporaso !

This looks good to merge as-is. I have a list of mostly minor to-do items as a follow-up from this PR, just a bit of cleanup and building out the unit tests etc, here: https://github.com/caporaso-lab/q2-phylogenomics/issues/10

**Pull request merged**